infinispan-embedded/src/main/java/org/hibernate/ogm/datastore/infinispan/InfinispanDialect.java

```diff
@@ -228,6 +247,104 @@ public void forEachTuple( ModelConsumer consumer, TupleTypeContext tupleTypeCont
 		return (KeyProvider<EK, AK, ISK>) provider.getKeyProvider();
 	}
 
+	@Override
+	public ClosableIterator<Tuple> callStoredProcedure( String storedProcedureName, ProcedureQueryParameters queryParameters, TupleContext tupleContext ) {
```

Comment:
Maybe we can create a new class (stateless and thread safe) to delegate stored procedure invocation. 
Put in there callStoredProcedure, validate, execute, instantiate, setParams, execute, extractTuples, extractTuple, methos and ExceptionWrapper class. The collaborator class can be created in the constructor as final blank filed.